src/tools/rbd/action/Bench.cc

```diff
@@ -293,18 +336,36 @@ int execute(const po::variables_map &vm) {
     return r;
   }
 
-  r = do_bench_write(image, bench_io_size, bench_io_threads, bench_bytes,
-                     bench_random);
+  r = do_bench(image, bench_io_type, bench_io_size, bench_io_threads,
+		     bench_bytes, bench_random);
   if (r < 0) {
-    std::cerr << "bench-write failed: " << cpp_strerror(r) << std::endl;
+    std::cerr << "do_bench failed: " << cpp_strerror(r) << std::endl;
     return r;
   }
   return 0;
 }
 
-Shell::Action action(
-  {"bench-write"}, {}, "Simple write benchmark.", "", &get_arguments, &execute);
+int execute_for_write(const po::variables_map &vm) {
+  return bench_execute(vm, BENCH_WRITE);
+}
+
+int execute_for_bench(const po::variables_map &vm) {
+  io_type_t bench_io_type;
+  if (vm.count("io-type")) {
+    bench_io_type = vm["io-type"].as<io_type_t>();
+  } else {
+    bench_io_type = BENCH_READ;
```

Comment:
Thanx @dillaman , yes, that's another confusion in my mind. If we want to make the size to be required, what about making it similar with qemu-img? E.g:
```
# qemu-img create test
qemu-img: test: Image creation needs a size parameter
# qemu-img create test 1G
Formatting 'test', fmt=raw size=1073741824 
```

Because ```--xxx``` gives me a feeling that it's optional and programmers have put an default value for my normal use case.

So can we do it in a similar way? such as:
```
# rbd bench read test
and 
# rbd bench write test
```

Both of them, ```rbd create```way and ```qemu-img create``` way, are acceptable to me, but I want to make sure it's a convention in our ceph world, so that if we meet another same situation, we can do it as what ```rbd create``` did. is that correct? @dillaman @trociny